File: BrownianMotion/parallel.py
from random import random, randint
from distance import ed_orig, md_orig
 
def experimento(dim, eucl):
    pos = [0] * dim
    mayor = 0
    for t in range(50): 
        cambiar = randint(0, dim - 1)
        cambio = 1 if random() < 0.5 else -1
        pos[cambiar] += cambio
        d = ed_orig(pos) if eucl else md_orig(pos)
        mayor = d if d > mayor else mayor
    return (eucl, dim, mayor)
 
import multiprocessing
if __name__ == "__main__":
    parametros = [(d, True) for d in range(1, 9)]
    parametros += [(d, False) for d in range(1, 9)]
    with multiprocessing.Pool() as pool:
        results = pool.starmap_async(experimento, parametros)
        print(results.get())

File: BrownianMotion/parallel.R
repetir <- 100
duracion <- 200
 
eucl <-  FALSE
library(parallel)
 
datos <-  data.frame()
 
experimento <- function(replica) {
    pos <- rep(0, dimension)
    mayor <- 0
    for (t in 1:duracion) {
        cambiar <- sample(1:dimension, 1)
        cambio <- 1
        if (runif(1) < 0.5) {
            cambio <- -1
        }
        pos[cambiar] <- pos[cambiar] + cambio
        if (eucl) {
            d <- sqrt(sum(pos**2))
        } else { # Manhattan
            d <- sum(abs(pos))
        }
        if (d > mayor) {
            mayor <- d
        }
    }
    return(mayor)
}
 
cluster <- makeCluster(detectCores() - 1)
clusterExport(cluster, "duracion")
clusterExport(cluster, "eucl")
clusterExport(cluster, "experimento")
 
for (dimension in 1:8) {
    clusterExport(cluster, "dimension")
    resultado <- parSapply(cluster, 1:repetir, experimento)
    datos <- rbind(datos, resultado)
}
stopCluster(cluster)
print(resultado)         
